### PR TITLE
BaanC Sections in functionlist.xml

### PR DESCRIPTION
**Pull request by oirfeodent**
Based on
https://notepad-plus-plus.org/community/topic/14494/functionlist-classrange-question

### CODE REVIEW

File: PowerEditor/src/functionList.xml [hunk 1]
```diff
@@ -1294,6 +1295,176 @@
...
+									:											# end-of-sub-section-header indicator
+								|
+									function									# keyword, start-of-function-header indicator
+									(?:(?&amp;COMMENT))?						# optional `embedded` comment
+									(?:											# optional storage-class specifier
+										\s+extern
+										(?:(?&amp;COMMENT))?					# ...with optional `embedded` comment
+									)?
+									(?:											# optional function type specifier
+										\s+(?:boolean
```
Comment (MAPJe71) [inline PowerEditor/src/functionList.xml]: Please put `boolean` on a new line with appropriate indentation (see "known sub-section names" above).

File: PowerEditor/src/functionList.xml [hunk 2]
```diff
@@ -1294,6 +1295,176 @@
...
+							^\h*												# optional leading whitespace at start-of-line
+							(?:
+								\K												# discard text matched so far
+								(?:												# known `empty` section names
+									after\.(?:form\.read|program|receive\.data|update\.db\.commit)
+								|	before\.(?:display\.object|new\.object|program)
+								|	declaration
+								|	on\.(?:display\.total\.line|error)
+								)
+								:												# end-of-section-header indicator
```
Comment (MAPJe71) [inline PowerEditor/src/functionList.xml]: likewise, see above.

File: PowerEditor/src/functionList.xml [hunk 3]
```diff
@@ -1294,6 +1295,176 @@
...
+							|
+								function										# keyword, start-of-function-header indicator
+								(?:(?&amp;COMMENT))?							# optional `embedded` comment
+								(?:												# optional storage-class specifier
+									\s+extern
+									(?:(?&amp;COMMENT))?						# ...with optional `embedded` comment
+								)?
+								(?:												# optional function type specifier
+									\s+(?:boolean
```
Comment (MAPJe71) [inline PowerEditor/src/functionList.xml]: likewise, see above.

File: PowerEditor/src/functionList.xml [hunk 4]
```diff
@@ -1294,6 +1295,176 @@
...
+										|	double
+										|	long
+										|	string
+										|	void
+										|	domain(?:(?&amp;COMMENT))?\s+[\w.]*	# Domain with optional `embedded` comment
```
Comment (MAPJe71) [inline PowerEditor/src/functionList.xml]: likewise, see above.

**Comment by oirfeodent** [comment]
@MAPJe71  
I have pushed another commit with the review changes.
Regards,

**Pull request closed without merging**